nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -118,6 +119,19 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
             # support datasets that are lists of lists
             collate_fn = dataset.datasets[0].datasets[0].collate_fn
 
+        if config.get('use_semi_sorted_batching', False):
```

Comment:
this can't be used with tarred datasets, so we should add a condition to check for that as well.

Reply:
This works as well.